include/detail/atomistic_keywords.hpp

```diff
@@ -292,62 +176,21 @@ namespace keywords
     struct cur_basis_line_value_z : pegtl::pad<decimal_number, pegtl::blank>
     { };
 
-    struct basis_value_line : pegtl::seq< pegtl::string<'#'>, pegtl::opt<TAO_PEGTL_ISTRING("#%")>, cur_basis_line_value_x, cur_basis_line_value_y, cur_basis_line_value_z>
+    struct basis_value_line : Vector3::vec3
     { };
 
-    struct basis_value : pegtl::seq< pegtl::eol, pegtl::list<basis_value_line, pegtl::eol > > 
+    struct basis_value : pegtl::seq< pegtl::eol, pegtl::list<pegtl::seq< pegtl::string<'#'>, pegtl::opt<TAO_PEGTL_ISTRING("#%")>, basis_value_line>, pegtl::eol > > 
     { };
 
-    template<>
-    struct kw_action< cur_basis_line_value_x >
-    {
-        template< typename Input >
-        static void apply( const Input& in, ovf_file & f, ovf_segment & segment)
-        {
-            if( !f._state->found_ncellpoints ) // Need to make sure that the basis array is already allocated
-            {
-                throw tao::pegtl::parse_error(fmt::format("ncellpoints must be specified before the basis!"), in);
-            }
-            segment.basis[3 * f._state->_cur_basis_line + 0] = std::stof(in.string());
-        }
-    };
-
-    template<>
-    struct kw_action< cur_basis_line_value_y >
-    {
-        template< typename Input >
-        static void apply( const Input& in, ovf_file & f, ovf_segment & segment)
-        {
-            if( !f._state->found_ncellpoints ) // Need to make sure that the basis array is already allocated
-            {
-                throw tao::pegtl::parse_error(fmt::format("ncellpoints must be specified before the basis!"), in);
-            }
-            segment.basis[3*f._state->_cur_basis_line + 1] = std::stof(in.string());
-        }
-    };
-
-    template<>
-    struct kw_action< cur_basis_line_value_z >
-    {
-        template< typename Input >
-        static void apply( const Input& in, ovf_file & f, ovf_segment & segment)
-        {
-            if( !f._state->found_ncellpoints ) // Need to make sure that the basis array is already allocated
-            {
-                throw tao::pegtl::parse_error(fmt::format("ncellpoints must be specified before the basis!"), in);
-            }
-            segment.basis[3* f._state->_cur_basis_line + 2] = std::stof(in.string());
-        }
-    };
-
     template<>
     struct kw_action< basis_value_line >
     {
         template< typename Input >
         static void apply( const Input& in, ovf_file & f, ovf_segment & segment)
         {
-            // fmt::print("basis line value: {}\n", in.string());
-            f._state->_cur_basis_line++;
+            float temp[3];
+            read_vector( in, temp );
+            f._state->_basis.push_back( {temp[0], temp[1], temp[2]} );
```

Comment:
https://github.com/spirit-code/ovf/blob/32da6aa6708063b275da9f76fe2b1cb3fe399088/include/detail/pegtl_defines.hpp#L61

I therefore believe it would be more canonical to write
```c++
auto atom = f._state->_basis.emplace_back();
read_vector( in, (*atom).data() );
```
or
```c++
std::array<float,3> atom;
read_vector( in, atom.data() );
f._state->_basis.push_back(atom);
```